### PR TITLE
WINDUPRULE-314, WINDUPRULE--315,WINDUPRULE-316 Resteasy new rules

### PR DESCRIPTION
**Pull request by mareknovotny**
https://issues.jboss.org/browse/WINDUPRULE-307

### CODE REVIEW

File: rules-reviewed/eap7/eap6/resteasy.windup.xml [hunk 1]
```diff
@@ -737,6 +737,93 @@
...
+                </or>
+            </when>
+            <perform>
+                <hint title="RESTEasy Text default charset response changed" effort="1" category-id="optional">
+                    <message>Resteasy should use UTF-8 as the character set for text media types.
+                    The `resteasy.add.charset` parameter is set to `true` by default. You can set the `resteasy.add.charset` parameter to false in web.xml `context-param` element.
+                    If you do not want RESTEasy to add `charset=UTF-8` to the returned content-type header when the resource method returns a `text/*` or `application/xml*` media type without an explicit charset. 
+                    </message>
+                    <link title="RESTEasy Default Charset UTF-8 in Content-Type Header"
+                        href="https://doc-stage.usersys.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.1/html-single/migration_guide/#migrate_resteasy_default_charset_utf8_in_content_type_header"/>
```
Comment (mrizzi) [inline rules-reviewed/eap7/eap6/resteasy.windup.xml]: Better replace this URL to not use `stages` servers ;)

File: rules-reviewed/eap7/eap6/resteasy.windup.xml [hunk 2]
```diff
@@ -737,6 +737,93 @@
...
+                        <location>ANNOTATION</location>
+                        <annotation-literal name="value" pattern="text/{*}"/>
+                    </javaclass>
+                </or>
+            </when>
+            <perform>
+                <hint title="RESTEasy Text default charset response changed" effort="1" category-id="optional">
+                    <message>Resteasy should use UTF-8 as the character set for text media types.
+                    The `resteasy.add.charset` parameter is set to `true` by default. You can set the `resteasy.add.charset` parameter to false in web.xml `context-param` element.
+                    If you do not want RESTEasy to add `charset=UTF-8` to the returned content-type header when the resource method returns a `text/*` or `application/xml*` media type without an explicit charset.
```
Comment (mrizzi) [inline rules-reviewed/eap7/eap6/resteasy.windup.xml]: Maybe it's better w/o a new sentence starting "If" but have something like:
_You can set the `resteasy.add.charset` parameter to false in web.xml `context-param` element if you do not want RESTEasy to add `charset=UTF-8` to the returned content-type header_ ...

File: rules-reviewed/eap7/eap6/resteasy.windup.xml [hunk 3]
```diff
@@ -737,6 +737,93 @@
...
+                    </perform>
+                </iteration>
+            </perform>
+        </rule>
+        <rule id="resteasy-eap6-000141">
+            <when>
+                <javaclass references="javax.ws.rs.Produces">
+                    <location>ANNOTATION</location>
+                    <annotation-literal name="value" pattern="application/x-java-serialized-object"/>
+                </javaclass>
```
Comment (mrizzi) [inline rules-reviewed/eap7/eap6/resteasy.windup.xml]: There's no condition for the `@Consumes(value="application/x-java-serialized-object")` case while it was enlisted in the JIRA [WINDUPRULE-316](https://issues.jboss.org/browse/WINDUPRULE-316).

Comment (mareknovotny) [reply rules-reviewed/eap7/eap6/resteasy.windup.xml]: good catch, I will fix it

File: rules-reviewed/eap7/eap6/resteasy.windup.xml (hunk 1 above)
Comment (mareknovotny) [reply rules-reviewed/eap7/eap6/resteasy.windup.xml]: sure, i will paste there 7.1Beta doc link instead.

File: rules-reviewed/eap7/eap6/resteasy.windup.xml (hunk 2 above)
Comment (mareknovotny) [reply rules-reviewed/eap7/eap6/resteasy.windup.xml]: ok, I'll try to reword the whole message, it seems like that if you copy&paste from more resources :(

**Pull request merged**